segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -226,104 +169,96 @@ public void updateSegmentAttribute(UpdateSegmentAttribute updateSegmentAttribute
-            }
-            return null;
-        }).exceptionally((Throwable e) -> {
-            handleException(requestId, segmentName, "Get attribute", e);
-            return null;
-        });
+        segmentStore.getStreamSegmentInfo(segmentName, false, TIMEOUT)
+                .thenAccept(properties -> {
+                    if (properties == null) {
+                        connection.send(new NoSuchSegment(requestId, segmentName));
+                    } else {
+                        Map<UUID, Long> attributes = properties.getAttributes();
+                        Long value = attributes.get(attributeId);
+                        if (value == null) {
+                            value = WireCommands.NULL_ATTRIBUTE_VALUE;
+                        }
+                        connection.send(new SegmentAttribute(requestId, value));
+                    }
+                })
+                .exceptionally(e -> handleException(requestId, segmentName, "Get attribute", e));
     }
-    
+
     @Override
     public void getStreamSegmentInfo(GetStreamSegmentInfo getStreamSegmentInfo) {
         String segmentName = getStreamSegmentInfo.getSegmentName();
-        CompletableFuture<SegmentProperties> future = segmentStore.getStreamSegmentInfo(segmentName, false, TIMEOUT);
-        future.thenApply(properties -> {
-            if (properties != null) {
-                StreamSegmentInfo result = new StreamSegmentInfo(getStreamSegmentInfo.getRequestId(),
-                                                                 properties.getName(),
-                                                                 true,
-                                                                 properties.isSealed(),
-                                                                 properties.isDeleted(),
-                                                                 properties.getLastModified().getTime(),
-                                                                 properties.getLength());
-                log.trace("Read stream segment info: {}", result);
-                connection.send(result);
-            } else {
-                log.trace("getStreamSegmentInfo could not find segment {}", segmentName);
-                connection.send(new StreamSegmentInfo(getStreamSegmentInfo.getRequestId(), segmentName, false, true, true, 0, 0));
-            }
-            return null;
-        }).exceptionally((Throwable e) -> {
-            handleException(getStreamSegmentInfo.getRequestId(), segmentName, "Get segment info", e);
-            return null;
-        });
+        segmentStore.getStreamSegmentInfo(segmentName, false, TIMEOUT)
+                .thenAccept(properties -> {
+                    if (properties != null) {
+                        StreamSegmentInfo result = new StreamSegmentInfo(getStreamSegmentInfo.getRequestId(),
+                                properties.getName(),
```

Comment:
If you're going to format something with one item per line, than indent on column so it's clear where they go.

Where I've ended up with my settings:
I've settled on function arguments being wrapped as needed but indented on column and chained invocations going on each on their own line if wrapping is needed at all. In both cases using column indent. I always use column indent except for constructor args, where I find they end up way off to the end of the line, so I use default indentation and wrap as needed. 
(As you can see this file was written before I settled on these settings)